awx_collection/plugins/modules/tower_organization.py

```diff
@@ -66,49 +77,61 @@
     tower_config_file: "~/tower_cli.cfg"
 '''
 
-from ..module_utils.ansible_tower import TowerModule, tower_auth_config, tower_check_mode
-
-try:
-    import tower_cli
-    import tower_cli.exceptions as exc
-
-    from tower_cli.conf import settings
-except ImportError:
-    pass
+from ..module_utils.tower_api import TowerModule
 
 
 def main():
+    # Any additional arguments that are not fields of the item can be added here
     argument_spec = dict(
-        name=dict(required=True),
-        description=dict(),
+        name=dict(type='str', required=True),
+        description=dict(type='str', required=False),
         custom_virtualenv=dict(type='str', required=False),
-        state=dict(choices=['present', 'absent'], default='present'),
+        max_hosts=dict(type='str', required=False, default="0"),
```

Comment:
why not just make the type here integer and then get rid of the custom logic to enforce that?

Reply:
We probably should do this. In an older release of Ansible I had problems getting an integer field into a module because it would find a string and die. However, it looks like this has been fixed somewhere along the line.